### PR TITLE
CSCQVAIN-253: Qvain UI facelift and clean up

### PR DESCRIPTION
**Pull request by jppiiroinen**

### CODE REVIEW

**Comment by jppiiroinen** [comment]
Screenshot 2019-07-03 at 20 57 00

**Comment by jppiiroinen** [comment]
Screenshot 2019-07-03 at 20 59 10
Screenshot 2019-07-03 at 20 58 33

**Comment by jppiiroinen** [comment]
this needs to be rebased with *next* branch and the tests needs to be updated as well.

File: src/widgets/Array.vue [hunk 1]
```diff
@@ -151,7 +155,7 @@ export default {
 				return `#${index +1} (Person)`
 			}
 
-			if (tabObjectType === 'Organization' && (tabObject.name['fi'] || tabObject.name['en'])) {
+			if (tabObjectType === 'Organization' && (tabObject && (tabObject.name['fi'] || tabObject.name['en']))) {
```
Comment (jppiiroinen) [inline src/widgets/Array.vue]: here is the patch needed for the tabObject

**Comment by jppiiroinen** [comment]
DateRange needs more work..

**Comment by jppiiroinen** [comment]
There was a change in the Date and DateRange components. Played with the concept of having inline versions, but the actual data input would get very difficult when you would have a lot of them within different years. The current version has free-form entry and it changes the styling, also set the Monday to be the first day of the week. The user facing date format has been changed.

In the case that the user faced date format is a problematic for backend integration we need to fix it in different PR, as we need to support different time and date formats depending on the selected UI language, for which we must implement the support in the future.

File: package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     "@fortawesome/vue-fontawesome": "^0.1.6",
     "autosize": "^4.0.2",
     "axios": "^0.19.0",
-    "bootstrap-vue": "^2.0.0-rc.15",
+    "bootstrap-vue": "^2.0.0-rc.25",
```
Comment (Kauhia) [inline package.json:28]: RC? are we using some release candidate version?

File: src/components/DateRange.vue [hunk 3]
```diff
@@ -58,13 +79,23 @@ export default {
...
+		},
 	},
 	methods: {
 		updateValue() {
 			this.$store.commit('updateValue', { p: this.parent, prop: this.property, val: {
 				start_date: this.startDateISO, end_date: this.endDateISO,
 			}})
 		},
+		format: function(value, event) {
+			console.log(value)
```
Comment (Kauhia) [inline src/components/DateRange.vue]: console.log

File: src/partials/DeleteButton.vue [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
+		<font-awesome-icon icon="trash" fixed-width class="icon" />
 	</span>
 </template>
 
 <style lang="scss" scoped>
 .delete-button {
 	padding: 4px 4px;
 	border-radius: 4px;
+	margin-left: 1em;
+	float: right;
```
Comment (Kauhia) [inline src/partials/DeleteButton.vue]: Not sure what this does in this case but I usually try to do layouting in the parent component. If this is correcting layout inside this component, ignore this message.

**Review by Kauhia — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (jppiiroinen) [reply package.json:28]: it looks like so..

File: src/components/DateRange.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/components/DateRange.vue]: meh that should not be there

File: package.json (hunk 2 above)
Comment (tahme) [reply package.json:28]: I think they just don't want to release a non-rc version or something :D 
v2.0.0-rc.0
Released: 2018-01-23

File: src/partials/DeleteButton.vue (hunk 4 above)
Comment (jppiiroinen) [reply src/partials/DeleteButton.vue]: true i will check a better location for those.

Comment (jppiiroinen) [reply src/partials/DeleteButton.vue]: should be fixed

File: src/components/DateRange.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/components/DateRange.vue]: should be fixed

File: src/components/Date.vue [hunk 5]
```diff
@@ -2,18 +2,30 @@
...
+				<datepicker
+					:inline="false"
+					placeholder="Select Date"
+					monday-first
+					bootstrapStyling
+					typeable
+					format="dd.MM.yyyy"
+					:id="property + '_date'"
 					v-model="date">
+					<b-button
```
Comment (Kauhia) [inline src/components/Date.vue]: Could a delete button be used here?

File: src/components/DateRange.vue [hunk 6]
```diff
@@ -1,16 +1,58 @@
...
+					:highlighted="highlightedSelected"
+					:inline="false"
+					bootstrapStyling
+					monday-first
+					typeable
+					placeholder="From"
+					format="dd.MM.yyyy"
+					:disabledDates="disableBefore"
+					v-model="start">
+					<b-button
```
Comment (Kauhia) [inline src/components/DateRange.vue]: Delete button?

File: src/components/DateRange.vue [hunk 7]
```diff
@@ -1,16 +1,58 @@
...
+					:highlighted="highlightedSelected"
+					:inline="false"
+					monday-first
+					bootstrapStyling
+					typeable
+					format="dd.MM.yyyy"
+					placeholder="To"
+					:disabledDates="disableAfter"
+					v-model="end">
+					<b-button
```
Comment (Kauhia) [inline src/components/DateRange.vue]: Delete button?

File: src/components/ReferenceData.vue [hunk 8]
```diff
@@ -281,6 +285,12 @@ export default {
 			this.getAllReferenceData()
 		}
 	},
+	beforeUpdate: function() {
```
Comment (Kauhia) [inline src/components/ReferenceData.vue:288]: Wouldn't created and mounted hooks work?

File: src/components/i18nTextarea.vue [hunk 9]
```diff
@@ -130,27 +115,28 @@ export default {
...
 			this.$set(this.state, lang, '')
-			this.updateValue()
-			this.focusOnLastTab()
 		},
 		deleteLang(lang) {
 			this.$delete(this.state, lang)
-			this.updateValue()
 		},
 		updateValue() {
+			console.log("update value called i18ntextarea")
```
Comment (Kauhia) [inline src/components/i18nTextarea.vue]: console.log

File: src/views/Editor.vue [hunk 10]
```diff
@@ -233,11 +295,15 @@ export default {
 				const isExisting = (currentId && currentId !== 'new')
 				if (isExisting) {
 					payload.id = currentId
-					const response = await apiClient.put("/datasets/" + currentId, payload)
-
+					const _ = await apiClient.put("/datasets/" + currentId, payload)
```
Comment (Kauhia) [inline src/views/Editor.vue]: No need for const _ =, should be fine with just await promise

File: src/widgets/String.vue [hunk 11]
```diff
@@ -2,60 +2,41 @@
...
+				<b-input-group>
+					<b-form-input
+						:id="inArray ? 'input-' + property.toString() : property"
+						:type="inputType"
+						:placeholder="uiPlaceholder"
+						:value="value"
+						:state="isValid ? null : false"
+						@input.native="updateValue">
+					</b-form-input>
+					<b-input-group-append>
```
Comment (Kauhia) [inline src/widgets/String.vue]: Delete button?

**Review by Kauhia — CHANGES_REQUESTED** [review]

File: src/components/i18nTextarea.vue (hunk 9 above)
Comment (jppiiroinen) [reply src/components/i18nTextarea.vue]: true.. used those to follow the updateValue triggers :D

File: src/components/ReferenceData.vue (hunk 8 above)
Comment (jppiiroinen) [reply src/components/ReferenceData.vue:288]: need to test it, afaik I looked the component lifecycle events yesterday, but I will verify and test it

File: src/views/Editor.vue (hunk 10 above)
Comment (jppiiroinen) [reply src/views/Editor.vue]: ok

File: src/widgets/String.vue (hunk 11 above)
Comment (jppiiroinen) [reply src/widgets/String.vue]: here is a clear button instead, not a delete.

File: src/components/DateRange.vue (hunk 7 above)
Comment (jppiiroinen) [reply src/components/DateRange.vue]: it is clear in this context, not delete

File: src/components/DateRange.vue (hunk 6 above)
Comment (jppiiroinen) [reply src/components/DateRange.vue]: it is clear in this context, not delete

File: src/components/DateRange.vue [hunk 12]
```diff
@@ -1,16 +1,58 @@
...
+						:id="property + '_date_clear-button'"
+						@click="end = null"
+						variant="outline-secondary"
+						class="vdp-datepicker_clear-button">
+						<font-awesome-icon icon="times" fixed-width class="icon" />
+					</b-button>
+				</datepicker>
+			</div>
+			<div class="col">
+				<delete-button v-if="inArray" @click="$emit('delete', property)"/>
```
Comment (jppiiroinen) [inline src/components/DateRange.vue:37]: here is the delete

File: src/widgets/String.vue [hunk 13]
```diff
@@ -2,60 +2,41 @@
...
+			{{ uiDescription }}
+		</small>
+
+		<div slot="input">
+			<b-form-group
+				:label-cols="inArray ? 1 : ((makeLabel !== uiLabel) ? 3 : 0)"
+				:label-for="inArray ? 'input-' + property.toString() : property">
+				<span slot="label" v-if="makeLabel !== uiLabel">
+					{{ makeLabel }}
+					<delete-button v-if="inArray" slot="label" @click="deleteMe"/>
```
Comment (jppiiroinen) [inline src/widgets/String.vue:15]: it is here the delete button

File: src/components/Date.vue (hunk 5 above)
Comment (jppiiroinen) [reply src/components/Date.vue]: this is a clear functionality not a delete here

Comment (jppiiroinen) [reply src/components/Date.vue]: removed

File: src/components/DateRange.vue (hunk 6 above)
Comment (jppiiroinen) [reply src/components/DateRange.vue]: removed

File: src/components/DateRange.vue (hunk 7 above)
Comment (jppiiroinen) [reply src/components/DateRange.vue]: removed

File: src/components/i18nTextarea.vue (hunk 9 above)
Comment (jppiiroinen) [reply src/components/i18nTextarea.vue]: removed

File: src/views/Editor.vue (hunk 10 above)
Comment (jppiiroinen) [reply src/views/Editor.vue]: removed

File: src/widgets/String.vue (hunk 11 above)
Comment (jppiiroinen) [reply src/widgets/String.vue]: removed

**Review by Kauhia — APPROVED** [review]

**Pull request merged**